impl/src/main/java/io/kroxylicious/testing/kafka/invm/ScramUtils.java

```diff
@@ -27,25 +36,272 @@ static ScramCredential asScramCredential(UserScramCredentialRecord uscr) {
         return new ScramCredential(uscr.salt(), uscr.storedKey(), uscr.serverKey(), uscr.iterations());
     }
 
-    @NonNull
-    static List<UserScramCredentialRecord> getScramCredentialRecords(String saslMechanism, Map<String, String> users) {
+    static List<UserScramCredentialRecord> getUserScramCredentialRecords(List<String> addScram) {
+        try {
+            return ScramParser.parse(addScram).stream()
+                    .map(ApiMessageAndVersion::message)
+                    .map(UserScramCredentialRecord.class::cast)
+                    .toList();
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Failed to create SCRAM users", e);
+        }
+    }
+
+    @NotNull
+    static List<String> toKafkaScramArguments(String saslMechanism, Map<String, String> users) {
         var scramMechanism = ScramMechanism.forMechanismName(saslMechanism);
         if (scramMechanism == null) {
             throw new IllegalArgumentException("unexpected scram mechanism " + saslMechanism);
         }
 
-        var addScram = users
+        return users
                 .entrySet()
                 .stream()
-                .map(e -> scramMechanism.mechanismName() + "=" + toKafkaScramCredentialsFormat(e.getKey(), e.getValue()))
+                .map(e -> scramMechanism.mechanismName() + "=" + toKafkaScramArguments(e.getKey(), e.getValue()))
                 .toList();
-
-        var n = new Namespace(Map.of("add_scram", addScram));
-        return CollectionConverters.asJava(StorageTool.getUserScramCredentialRecords(n).get());
     }
 
     @NonNull
-    private static String toKafkaScramCredentialsFormat(String username, String password) {
+    private static String toKafkaScramArguments(String username, String password) {
         return "[name=%s,password=%s]".formatted(username, password);
     }
+
+    /** Copied from Kafka 3.9 ScramParser */
```

Comment:
Verbatim copy of  ScramParser from Kafka 3.9.   The kafka class's accessibility mean we can't use it directly.